### PR TITLE
refactor(ui, core): delete_message

### PR DESCRIPTION
**Pull request by deven98**

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 1]
```diff
@@ -275,6 +277,12 @@ class MessageWidget extends StatefulWidget {
   /// Function called on long press
   final void Function(BuildContext, Message)? onMessageActions;
 
+  /// Widget builder for building a bottom row below the message when deleted
+  final Widget Function(BuildContext, Message)? bottomRowBuilder;
+
+  /// Widget builder for building a bottom row below the message when deleted
+  final Widget Function(BuildContext, Message)? deletedBottomRowBuilder;
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:284]: Same doc-comments :|

Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/message_widget.dart:284]:
```suggestion
  /// Widget builder for building a bottom row below the message
  final Widget Function(BuildContext, Message)? bottomRowBuilder;

  /// Widget builder for building a bottom row below a deleted message
  final Widget Function(BuildContext, Message)? deletedBottomRowBuilder;
```

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 2]
```diff
@@ -830,22 +847,11 @@ class _MessageWidgetState extends State<MessageWidget>
...
-            style: chatThemeData.textTheme.footnote
-                .copyWith(color: chatThemeData.colorTheme.textLowEmphasis),
-          ),
-        ],
-      );
+      return widget.deletedBottomRowBuilder?.call(
+            context,
+            widget.message,
+          ) ??
+          const SizedBox.shrink();
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:854]: considering using an Offstage, is there any problem with that?
```suggestion
          const Offstage();
```

**Review by imtoori — CHANGES_REQUESTED** [review]
minor fixes

File: packages/stream_chat_flutter/lib/src/message_widget.dart (hunk 2 above)
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:854]: this should be `VisibleFootnote` instead of `SizedBox`

Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/message_widget.dart:854]: nope, that's right
we're always showing delete messages

File: packages/stream_chat_flutter_core/lib/src/message_list_core.dart [hunk 3]
```diff
@@ -134,8 +134,7 @@ class MessageListCoreState extends State<MessageListCore> {
 
     bool defaultFilter(Message m) {
       final isMyMessage = m.user?.id == _currentUser?.id;
-      final isDeletedOrShadowed = m.isDeleted == true || m.shadowed == true;
-      if (isDeletedOrShadowed && !isMyMessage) return false;
+      if (m.shadowed && !isMyMessage) return false;
```
Comment (xsahil03x) [inline packages/stream_chat_flutter_core/lib/src/message_list_core.dart:137]: are we showing deleted messages to all the users? Or only for current user

Comment (imtoori) [reply packages/stream_chat_flutter_core/lib/src/message_list_core.dart:137]: all users

**Review by imtoori — APPROVED** [review]
LGTM

**Pull request merged**